Begin v2

pangperry opened the pull request:
I've removed the private method so I can temporarily test the otherwise private methods. I need to dig into rspec a bit more to see how I should be dealing with the methods I will call from within the encrypt_message method.

adarsh commented on spec/solitaire_spec.rb:
```diff
@@ -1,7 +1,29 @@
+    message = 'Code in Ruby, live longer!'
+    expected_output = ["CODEI", "NRUBY", "LIVEL", "ONGER"]
+
+    prepared_message  = Solitaire.new.prepare(message)
```
It's a bit more canonical to pass the `message` in the constructor as in `Solitaire.new(message).prepare`.

pangperry replied:
I'll change this. Thanks.

adarsh commented on spec/solitaire_spec.rb:
```diff
@@ -11,3 +33,4 @@
     expect(decrypted_message).to eq expected_output
   end
 end
+
```
Style: Extra newline.

pangperry replied:
whoopsie. good eye. I'll change this too, thanks. 🙏

adarsh commented:
> I've removed the private method so I can temporarily test the otherwise private methods.

In general, it's bad practice to test private methods. You are currently testing at a high level (e.g. input, encrypt/decrypt, assert on the output), which is totally fine and sufficient.

You could add more specific "unit" level tests if you wanted to add more features, but you don't really need that at this point, since the end-to-end process is working as expected.

> I need to dig into rspec a bit more to see how I should be dealing with the methods I will call from within the encrypt_message method.

To reiterate, don't test private methods. :)

adarsh commented on solitaire.rb:
```diff
@@ -1,4 +1,35 @@
 class Solitaire
```
If you are starting over, why isn't this a new file? Or did I misunderstand the commit message?

adarsh commented on spec/solitaire_spec.rb:
```diff
@@ -1,7 +1,29 @@
+
+    expect(encrypted_message).to eq expected_output
+  end
+end
```
These tests look okay, although I would move the message argument into the constructor instead of as a parameter into the public method.

adarsh commented:
Couple of comments, but otherwise looks good.

pangperry committed:
Cleanup up for v1

*includes additional refactoring for final v1
*incorporates changes recommended by Adarsh
*key_deck removed. working replacement needed

move prepare method up to clean_message

Move substitute chars closer to where it's called

Move add_x closer to where it's called

Move convert_characters closer to where it's called

Move converter closer to where it's called

Move converter_hash closer to where it's called

Move convert_keystream_message closer to where it's called

Remove extra newline

Move message argument to constructor

Remove solitaire.rb content to prep for rewrite

Remove unused key_deck.
